vmware_aria_operations_integration_sdk/mp_build.py

```diff
@@ -108,48 +109,69 @@ def build_subdirectories(directory: str) -> None:
         shutil.move(os.path.join(directory, file), dir_path)
 
 
-def get_registry_components(docker_registry: str) -> Tuple[str, str]:
-    components = docker_registry.split("/")
+def get_registry_components(container_registry: str) -> Tuple[str, str]:
+    components = container_registry.split("/")
     host = components[0]
     path = "/".join(components[1:])
     return host, path
 
 
-def is_valid_registry(docker_registry: str, **kwargs: Any) -> bool:
+def is_valid_registry(container_registry: str, **kwargs: Any) -> bool:
     try:
-        login(docker_registry, **kwargs)
+        if _is_docker_hub_registry_format(container_registry):
+            if "registry_username" not in kwargs:
+                kwargs["registry_username"] = prompt("Enter Docker Hub username: ")
+
+            if "registry_password" not in kwargs:
+                kwargs["registry_password"] = prompt("Password: ", is_password=True)
+            login(**kwargs)
+            container_registry = f"docker.io/{container_registry}"
+
+        login(container_registry=container_registry, **kwargs)
+
     except LoginError:
         return False
 
     return True
 
 
+def _is_docker_hub_registry_format(registry: str) -> bool:
+    # if pattern for dockechub is username/repo
+    pattern = r"^[\w\-]+\/[\w\-]+$"
```

Comment:
`_` and `.` are also allowed. Also, should we check that the user hasn't added the full 'docker.io/namespace/repository' path?

Reply:
ah, you're right, it does match `_` -- however, `\w` also matches uppercase letters, which I believe aren't allowed.

Reply:
I'm not sure what you mean by 'isn't valid outside of the tag'. I can create a repository with `.`s in it:
![image](https://github.com/vmware/vmware-aria-operations-integration-sdk/assets/3310870/706e02f3-eebb-4cda-a200-68702b478ffe)